pandas/core/groupby/generic.py

```diff
@@ -1721,6 +1721,16 @@ def value_counts(
             in_axis_names = {
                 grouping.name for grouping in self.grouper.groupings if grouping.in_axis
             }
+
+            # adding optional subset handeling
+            if subset is None:
+                names = subset
+            else:
+                names = []
+                for name in self._selected_obj.columns:
+                    if name not in in_axis_names:
+                        names.append(name)
```

Comment:
Can you make this a set and use comprehension:

    names = {name for name in self._selected_obj.columns if name not in in_axis_names}